CCS/Guadaloop_lib/guadaloop/drivers/sensors/distance/AdafruitVL53L0X.c

```diff
@@ -1,12 +1,149 @@
 /*
  * AdafruitVL53L0X.c
  *
- *  Created on: Dec 31, 2022
- *      Author: dario
+ *  Created on: Oct 11, 2023
+ *      Author: Daniela
  */
 
 
 #include <guadaloop/drivers/sensors/distance/AdafruitVL53L0X.h>
+#include <stdint.h>
+#include <ti/devices/msp432e4/driverlib/driverlib.h>
+
+#define ONE_BYTE 1
+#define TWO_BYTES 2
+
+void VL6180x_setRegister(uint16_t registerAddr, uint8_t sendData, uint8_t byteSize);
+uint8_t VL6180x_getRegister(uint16_t registerAddr, uint8_t byteSize);
 
 //TODO
 
+uint8_t VL6180x_init(){
+   uint8_t data = VL6180x_getRegister(SYSTEM__FRESH_OUT_OF_RESET, ONE_BYTE);
+
+   if(data!=1){
+       //reset failed
+       return 0;
+   }
+
+   VL6180x_setRegister(0x0207, 0x01, ONE_BYTE);
+   VL6180x_setRegister(0x0208, 0x01, ONE_BYTE);
+   VL6180x_setRegister(0x0096, 0x00, ONE_BYTE);
+   VL6180x_setRegister(0x0097, 0xfd, ONE_BYTE);
+   VL6180x_setRegister(0x00e3, 0x01, ONE_BYTE);
+   VL6180x_setRegister(0x00e4, 0x03, ONE_BYTE);
+   VL6180x_setRegister(0x00e5, 0x02, ONE_BYTE);
+   VL6180x_setRegister(0x00e6, 0x01, ONE_BYTE);
+   VL6180x_setRegister(0x00e7, 0x03, ONE_BYTE);
+   VL6180x_setRegister(0x00f5, 0x02, ONE_BYTE);
+   VL6180x_setRegister(0x00d9, 0x05, ONE_BYTE);
+   VL6180x_setRegister(0x00db, 0xce, ONE_BYTE);
+   VL6180x_setRegister(0x00dc, 0x03, ONE_BYTE);
+   VL6180x_setRegister(0x00dd, 0xf8, ONE_BYTE);
+   VL6180x_setRegister(0x009f, 0x00, ONE_BYTE);
+   VL6180x_setRegister(0x00a3, 0x3c, ONE_BYTE);
+   VL6180x_setRegister(0x00b7, 0x00, ONE_BYTE);
+   VL6180x_setRegister(0x00bb, 0x3c, ONE_BYTE);
+   VL6180x_setRegister(0x00b2, 0x09, ONE_BYTE);
+   VL6180x_setRegister(0x00ca, 0x09, ONE_BYTE);
+   VL6180x_setRegister(0x0198, 0x01, ONE_BYTE);
+   VL6180x_setRegister(0x01b0, 0x17, ONE_BYTE);
+   VL6180x_setRegister(0x01ad, 0x00, ONE_BYTE);
+   VL6180x_setRegister(0x00ff, 0x05, ONE_BYTE);
+   VL6180x_setRegister(0x0100, 0x05, ONE_BYTE);
+   VL6180x_setRegister(0x0199, 0x05, ONE_BYTE);
+   VL6180x_setRegister(0x01a6, 0x1b, ONE_BYTE);
+   VL6180x_setRegister(0x01ac, 0x3e, ONE_BYTE);
+   VL6180x_setRegister(0x01a7, 0x1f, ONE_BYTE);
+   VL6180x_setRegister(0x0030, 0x00, ONE_BYTE);
+
+
+   //interrupt settings
+   VL6180x_setRegister(SYSTEM__INTERRUPT_CONFIG_GPIO, (4<<3)|3, ONE_BYTE);
+   VL6180x_setRegister(0x0011, 0x10, ONE_BYTE);
+   VL6180x_setRegister(0x010A, 0x30, ONE_BYTE);
+   VL6180x_setRegister(0x003f, 0x46, ONE_BYTE);
+   VL6180x_setRegister(0x0031, 0xFF, ONE_BYTE);
+   VL6180x_setRegister(0x0041, 0x63, ONE_BYTE);
+   VL6180x_setRegister(0x002e, 0x01, ONE_BYTE);
+
+   return 1;
+
+}
+
+uint8_t VL6180x_getDistance(){
+    VL6180x_setRegister(SYSRANGE__START,0x02, ONE_BYTE); //select single shot mode
+    while((VL6180x_getRegister(RESULT__INTERRUPT_STATUS_GPIO, ONE_BYTE) & 0x02)>>1 != 1){};
+    uint8_t asl_data = VL6180x_getRegister(RESULT__RANGE_VAL, ONE_BYTE); // 1 byte
+    VL6180x_setRegister(SYSTEM__INTERRUPT_CLEAR,0x07, ONE_BYTE);
+    return asl_data;
+}
+
+
+
+void VL6180x_initCommunication(){
```

Comment:
Is there any sensor specifics stuff in here? if not, maybe move this to another file

Reply:
I ended up removing this because we can initialize the pins we are going to use for I2C in the init.c files of the hub